packages/next/src/server/dev/hot-reloader-webpack.ts

```diff
@@ -379,13 +383,16 @@ export default class HotReloaderWebpack implements NextJsHotReloaderInterface {
 
     const { finished } = await handlePageBundleRequest(res, parsedUrl)
 
-    for (const fn of this.interceptors) {
-      await new Promise<void>((resolve, reject) => {
-        fn(req, res, (err: Error) => {
-          if (err) return reject(err)
-          resolve()
```

Comment:
Note that we were only resolving the promise when `next()` was called, which the middleware only does if it doesn't handle a request.